src/measure_theory/measure/finite_measure_weak_convergence.lean

```diff
@@ -154,6 +158,189 @@ def coe_add_monoid_hom : finite_measure α →+ measure α :=
+end
+
+-- Only useful here or more generally?
+-- Where to place?
+lemma _root_.bounded_continuous_function.nnreal.to_ennreal_comp_measurable {α : Type*}
+  [topological_space α] [measurable_space α] [opens_measurable_space α] (f : α →ᵇ ℝ≥0) :
+  measurable (λ x, (f x : ℝ≥0∞)) :=
+measurable_coe_nnreal_ennreal.comp f.continuous.measurable
+
+-- This does not seem unreasonable to me, although it may be a bit specific.
+-- Where to place?
+lemma bounded_continuous_function.nnreal.upper_bound {α : Type*} [topological_space α]
+  (f : α →ᵇ ℝ≥0) (x : α) : f x ≤ nndist f 0 :=
+begin
+  have key' : nndist (f x) ((0 : α →ᵇ ℝ≥0) x) ≤ nndist f 0,
+  { exact @bounded_continuous_function.dist_coe_le_dist α ℝ≥0 _ _ f 0 x, },
+  simp only [bounded_continuous_function.coe_zero, pi.zero_apply] at key',
+  rwa nnreal.nndist_zero_eq_val' (f x) at key',
+end
+
+-- This is the formulation I prefer in the present context, naturally uses the more general ones.
+lemma lintegral_lt_top_of_bounded_continuous_to_nnreal (μ : finite_measure α) (f : α →ᵇ ℝ≥0) :
+  ∫⁻ x, f x ∂(μ : measure α) < ∞ :=
+begin
+  apply is_finite_measure.lintegral_lt_top_of_bounded_to_ennreal,
+  use nndist f 0,
+  intros x,
+  have key := bounded_continuous_function.nnreal.upper_bound f x,
+  rw ennreal.coe_le_coe,
+  have eq : nndist f 0 = ⟨dist f 0, dist_nonneg⟩,
+  { ext,
+    simp only [real.coe_to_nnreal', max_eq_left_iff, subtype.coe_mk, coe_nndist], },
+  rwa eq at key,
+end
+
+@[simp] lemma test_against_nn_coe_eq {μ : finite_measure α} {f : α →ᵇ ℝ≥0} :
+  (μ.test_against_nn f : ℝ≥0∞) = ∫⁻ x, f x ∂(μ : measure α) :=
+ennreal.coe_to_nnreal (lintegral_lt_top_of_bounded_continuous_to_nnreal μ f).ne
+
+lemma test_against_nn_const (μ : finite_measure α) (c : ℝ≥0) :
+  μ.test_against_nn (bounded_continuous_function.const α c) = c * μ.mass :=
+by simp [← ennreal.coe_eq_coe]
+
+lemma test_against_nn_mono (μ : finite_measure α)
+  {f g : α →ᵇ ℝ≥0} (f_le_g : (f : α → ℝ≥0) ≤ g) :
+  μ.test_against_nn f ≤ μ.test_against_nn g :=
+begin
+  simp only [←ennreal.coe_le_coe, test_against_nn_coe_eq],
+  apply lintegral_mono,
+  exact λ x, ennreal.coe_mono (f_le_g x),
+end
+
+variables [opens_measurable_space α]
+
+lemma test_against_nn_add (μ : finite_measure α) (f₁ f₂ : α →ᵇ ℝ≥0) :
+  μ.test_against_nn (f₁ + f₂) = μ.test_against_nn f₁ + μ.test_against_nn f₂ :=
+begin
+  simp only [←ennreal.coe_eq_coe, bounded_continuous_function.coe_add, ennreal.coe_add,
+             pi.add_apply, test_against_nn_coe_eq],
+  apply lintegral_add,
+  repeat { exact (bounded_continuous_function.nnreal.to_ennreal_comp_measurable _), },
+end
+
+-- Hmmm, what is the right way to do this `nnreal_mul_ennreal_to_nnreal`?
+-- If this deserves to be added, then perhaps in `data.real.ennreal`?
+-- It seems like a typical coercion issue to me, although it is only used once here.
+lemma _root_.ennreal.nnreal_mul_ennreal_to_nnreal (c : ℝ≥0) (z : ℝ≥0∞) :
+  c * z.to_nnreal = ((c : ℝ≥0∞) * z).to_nnreal :=
```

Comment:
You no longer need this for the next lemma!  Maybe we have found the tricks to avoid it ever being needed ... do you want to remove it at least from this PR?